docs/content/concepts/configuration/config-schema.mdx

```diff
@@ -15,69 +15,75 @@ Run configuration allows providing parameters to jobs at the time they're execut
 
 ## Overview
 
-It's often useful to configure jobs at run time. For example, you might want someone to manually operate a deployed job and choose what dataset it operates on when they run it. In general, you should use Dagster's config system when you want the person or software that is executing a job to be able to make choices about what the job does, without needing to modify the job definition.
+It's often useful to configure Dagster jobs at runtime. For example, you might want to choose what dataset an op runs against, or provide a connection URL for a database resource. In general, you should use Dagster's config system when you want to modulate a computation at runtime without needing to modify the definition.
 
-The objects that compose a job - ops and resources - are each individually configurable. When executing a job, you can supply "run configuration" that specifies the configuration for each of the objects in the job. When you execute a job with Dagster's Python API, you supply run configuration as a Python dictionary. When you execute a job from Dagit or the CLI, you can provide config in a YAML document.
+Various Dagster entities (ops, assets, resources) can have an attached _configuration schema_ that specifies the configurable parameters for that entity. When launching a job that executes (ops), materializes (assets), or instantiates (resources) a configurable entity, you can provide values for the parameters specified in the schema. Dagster validates this _run configuration_ against the schema and proceeds only if validation is successful. Within the function that defines the entity, you can access the passed-in configuration off of the `context`.
```

Comment:
Not leading with config schema was a conscious choice from a prior revision - we were concerned that asking readers to immediately understand the notion of config schema would make the system less approachable.  I don't think that we should try to bury config schema, but I think the notion that ops, software-defined assets, and resources can be configured should be introduced before introducing schema.

Reply:
The impetus to put schema first came from the fact that you cannot use config in assets without schema (this is a consequence of there already being an implicit schema with the `assets` key for assets). If we want to present assets and ops together (which IMO we should for simplicity's sake), the old ordering becomes very awkward since the first section (accessing config inside compute fn _without_ schema) does not work for assets.

FWIW however, I also find the new ordering more clear generally, especially if we are discouraging config without schema. IMO 99% of the time, people are going to think: "Configuration? Cool, first, where do I define the accepted parameters? Second, how do I access them?" Not: "First, how do I access the parameters? Second, how do I specify them?" It's like you'd typically write a function signature before writing the body.

>we were concerned that asking readers to immediately understand the notion of config schema would make the system less approachable.

I agree that attempting a full presentation of config schema would be undesirable, but the common case (which is used in the example) is simple and virtually self-explanatory: key-value pairs with keys parameter names and values simple types. I mentioned that I have some other improvements for the page in mind, one of these is a greater-depth explanation of schema further down.

>I think the notion that ops, software-defined assets, and resources can be configured should be introduced before introducing schema.

Is this not already introduced beforehand the Overview section?

Reply:
Hmm actually now I see that you are commenting on the Overview, I thought your comment was about the flow of the lower sections. It's an easy fix in the Overview.